tel2puml/utils.py:77: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/utils.py:77: error: Missing type parameters for generic type "tuple"  [type-arg]
tel2puml/utils.py:78: error: Name "plt.Figure" is not defined  [name-defined]
tel2puml/utils.py:108: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/utils.py:142: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/sequence_otel.py:14: error: Untyped decorator makes function "get_slice_indexes" untyped  [misc]
tel2puml/sequence_otel.py:31: error: Untyped decorator makes function "has_time_overlap" untyped  [misc]
tel2puml/sequence_otel.py:190: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/sequence_otel.py:248: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/sequence_otel.py:273: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/sequence_otel.py:292: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/sequence_otel.py:322: error: Need type annotation for "_graph"  [var-annotated]
tel2puml/detect_loops.py:89: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/detect_loops.py:210: error: Incompatible types in assignment (expression has type "list[Loop] | list[str]", variable has type "list[Loop]")  [assignment]
tel2puml/detect_loops.py:467: error: Incompatible types in assignment (expression has type "list[Loop] | list[str]", variable has type "list[str]")  [assignment]
tel2puml/detect_loops.py:557: error: Need type annotation for "break_edges" (hint: "break_edges: List[<type>] = ...")  [var-annotated]
tel2puml/detect_loops.py:788: error: Value of type "OutEdgeDataView[T, tuple[T, T]]" is not indexable  [index]
tel2puml/detect_loops.py:804: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/detect_loops.py:821: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/check_puml_equiv.py:54: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/check_puml_equiv.py:112: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/check_puml_equiv.py:125: error: Need type annotation for "graph"  [var-annotated]
tel2puml/check_puml_equiv.py:130: error: Need type annotation for "collected_attributes" (hint: "collected_attributes: Dict[<type>, <type>] = ...")  [var-annotated]
tel2puml/check_puml_equiv.py:220: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/check_puml_equiv.py:229: error: Argument 1 to "create_networkx_graph_from_parsed_puml" has incompatible type "list[str]"; expected "list[Any | tuple[Literal['START', 'END', 'PATH'], Literal['XOR', 'AND', 'OR', 'LOOP']]]"  [arg-type]
tel2puml/check_puml_equiv.py:229: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
tel2puml/check_puml_equiv.py:229: note: Consider using "Sequence" instead, which is covariant
tel2puml/check_puml_equiv.py:270: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/check_puml_equiv.py:281: error: Returning Any from function declared to return "bool"  [no-any-return]
tel2puml/check_puml_equiv.py:308: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/check_puml_equiv.py:309: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/check_puml_equiv.py:330: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tests/tel2puml/test_sequence_otel.py:99: error: Need type annotation for "graph"  [var-annotated]
tests/tel2puml/test_sequence_otel.py:101: error: Non-overlapping equality check (left operand type: "OutEdgeView[Any]", right operand type: "set[tuple[Span, Span]]")  [comparison-overlap]
tests/tel2puml/test_sequence_otel.py:113: error: Non-overlapping equality check (left operand type: "OutEdgeView[Any]", right operand type: "set[tuple[Span, Span]]")  [comparison-overlap]
tests/tel2puml/test_sequence_otel.py:127: error: Need type annotation for "graph"  [var-annotated]
tests/tel2puml/test_sequence_otel.py:158: error: Argument 1 to "add_span" of "Trace" has incompatible type "*tuple[object, ...]"; expected "str"  [arg-type]
tests/tel2puml/test_sequence_otel.py:158: error: Argument 1 to "add_span" of "Trace" has incompatible type "*tuple[object, ...]"; expected "float"  [arg-type]
tests/tel2puml/test_sequence_otel.py:158: error: Argument 1 to "add_span" of "Trace" has incompatible type "*tuple[object, ...]"; expected "str | None"  [arg-type]
tests/tel2puml/test_check_puml_equiv.py:33: error: Need type annotation for "collected_attributes" (hint: "collected_attributes: Dict[<type>, <type>] = ...")  [var-annotated]
tests/tel2puml/test_check_puml_equiv.py:104: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
tests/tel2puml/test_check_puml_equiv.py:213: error: Need type annotation for "graph"  [var-annotated]
tests/tel2puml/test_check_puml_equiv.py:217: error: Dict entry 0 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/tel2puml/test_check_puml_equiv.py:218: error: Dict entry 1 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/tel2puml/test_check_puml_equiv.py:221: error: Dict entry 0 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/tel2puml/test_check_puml_equiv.py:222: error: Dict entry 1 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/tel2puml/test_check_puml_equiv.py:253: error: Need type annotation for "graph"  [var-annotated]
tests/tel2puml/test_check_puml_equiv.py:257: error: Dict entry 0 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/tel2puml/test_check_puml_equiv.py:258: error: Dict entry 1 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/tel2puml/test_check_puml_equiv.py:261: error: Dict entry 0 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/tel2puml/test_check_puml_equiv.py:262: error: Dict entry 1 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/tel2puml/test_check_puml_equiv.py:285: error: Incompatible types in assignment (expression has type "bool", target has type "str")  [assignment]
tests/tel2puml/test_check_puml_equiv.py:286: error: Incompatible types in assignment (expression has type "bool", target has type "str")  [assignment]
tel2puml/logic_detection.py:396: error: Argument 1 to "remove" of "set" has incompatible type "set[Any]"; expected "frozenset[str]"  [arg-type]
tel2puml/logic_detection.py:398: error: Argument 1 to "get_weighted_cover" has incompatible type "set[frozenset[str]]"; expected "set[set[Any] | frozenset[Any]]"  [arg-type]
tel2puml/events.py:43: error: Signature of "__hash__" incompatible with supertype "dict"  [override]
tel2puml/events.py:43: note:      Superclass:
tel2puml/events.py:43: note:          None
tel2puml/events.py:43: note:      Subclass:
tel2puml/events.py:43: note:          def __hash__(self) -> int
tel2puml/walk_puml_graph/node.py:75: error: Unsupported left operand type for + ("None")  [operator]
tel2puml/walk_puml_graph/node.py:75: note: Left operand is of type "str | None"
tel2puml/walk_puml_graph/node.py:331: error: Incompatible return value type (got "tuple[PUMLEvent, ...]", expected "tuple[PUMLEvent]")  [return-value]
tel2puml/walk_puml_graph/node.py:332: error: Incompatible return value type (got "tuple[Never, ...]", expected "tuple[PUMLEvent]")  [return-value]
tel2puml/walk_puml_graph/node.py:568: error: Need type annotation for "networkx_graph"  [var-annotated]
tel2puml/walk_puml_graph/node.py:602: error: Need type annotation for "event_node_ref" (hint: "event_node_ref: Dict[<type>, <type>] = ...")  [var-annotated]
tests/tel2puml/test_logic_detection.py:65: error: Argument 1 to "create_augmented_data_from_reduced_event_set" has incompatible type "EventSet"; expected "frozenset[str]"  [arg-type]
tests/tel2puml/test_logic_detection.py:221: error: Function is missing a type annotation  [no-untyped-def]
tests/tel2puml/test_logic_detection.py:240: error: Function is missing a type annotation  [no-untyped-def]
tests/tel2puml/test_logic_detection.py:256: error: Call to untyped function "_check_or" in typed context  [no-untyped-call]
tests/tel2puml/test_logic_detection.py:263: error: Call to untyped function "_check_or_and" in typed context  [no-untyped-call]
tests/tel2puml/test_logic_detection.py:453: error: Function is missing a return type annotation  [no-untyped-def]
tests/tel2puml/test_logic_detection.py:543: error: Function is missing a type annotation  [no-untyped-def]
tests/tel2puml/test_logic_detection.py:552: error: Call to untyped function "_check_node_for_repeat_and" in typed context  [no-untyped-call]
tests/tel2puml/test_logic_detection.py:572: error: Call to untyped function "_check_node_for_repeat_and" in typed context  [no-untyped-call]
tests/tel2puml/test_detect_loops.py:509: error: Incompatible types in assignment (expression has type "list[Loop] | list[str]", variable has type "list[Loop]")  [assignment]
tests/tel2puml/test_detect_loops.py:531: error: Incompatible types in assignment (expression has type "list[Loop] | list[str]", variable has type "list[Loop]")  [assignment]
tests/tel2puml/test_detect_loops.py:831: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tests/tel2puml/test_detect_loops.py:833: error: Need type annotation for "graph"  [var-annotated]
tests/tel2puml/test_detect_loops.py:881: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tests/tel2puml/test_detect_loops.py:883: error: Need type annotation for "graph"  [var-annotated]
tests/tel2puml/test_detect_loops.py:911: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tests/tel2puml/test_detect_loops.py:913: error: Need type annotation for "graph"  [var-annotated]
tests/tel2puml/test_detect_loops.py:924: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tests/tel2puml/test_detect_loops.py:1004: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tests/tel2puml/test_detect_loops.py:1011: error: Need type annotation for "graph"  [var-annotated]
tests/tel2puml/test_detect_loops.py:1050: error: Need type annotation for "graph"  [var-annotated]
tests/tel2puml/walk_puml_graph/test_node.py:86: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/tel2puml/walk_puml_graph/test_node.py:110: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/tel2puml/walk_puml_graph/test_node.py:142: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/tel2puml/walk_puml_graph/test_node.py:174: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/tel2puml/walk_puml_graph/test_node.py:207: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/tel2puml/walk_puml_graph/test_node.py:255: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/tel2puml/walk_puml_graph/test_node.py:293: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/tel2puml/walk_puml_graph/test_node.py:328: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/tel2puml/walk_puml_graph/test_node.py:467: error: Argument 3 to "load_logic_tree_into_nodes" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/tel2puml/walk_puml_graph/test_node.py:495: error: Argument 3 to "load_all_logic_trees_into_nodes" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/tel2puml/walk_puml_graph/test_node.py:512: error: Need type annotation for "graph"  [var-annotated]
tests/tel2puml/walk_puml_graph/test_node.py:588: error: Need type annotation for "expected_incoming_event_type_logic"  [var-annotated]
tests/tel2puml/walk_puml_graph/test_node.py:605: error: Unsupported right operand type for in ("object")  [operator]
tests/tel2puml/walk_puml_graph/test_node.py:608: error: Value of type variable "SupportsRichComparisonT" of "sorted" cannot be "str | None"  [type-var]
tests/tel2puml/walk_puml_graph/test_node.py:614: error: Value of type "object" is not indexable  [index]
tests/tel2puml/walk_puml_graph/test_node.py:627: error: Value of type variable "SupportsRichComparisonT" of "sorted" cannot be "str | None"  [type-var]
tests/tel2puml/walk_puml_graph/test_node.py:638: error: Value of type variable "SupportsRichComparisonT" of "sorted" cannot be "str | None"  [type-var]
tests/tel2puml/walk_puml_graph/test_node.py:644: error: Value of type variable "SupportsRichComparisonT" of "sorted" cannot be "str | None"  [type-var]
tel2puml/puml_graph/graph.py:58: error: "type[PUMLOperatorNodes]" has no attribute "PATH_LOOP"  [attr-defined]
tel2puml/puml_graph/graph.py:81: error: Argument "extra_info" to "__init__" of "NXNode" has incompatible type "dict[str, bool] | None"; expected "dict[str, str] | None"  [arg-type]
tel2puml/puml_graph/graph.py:301: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/puml_graph/graph.py:397: error: Incompatible return value type (got "tuple[PUMLOperatorNode, ...]", expected "tuple[PUMLOperatorNode, PUMLOperatorNode]")  [return-value]
tel2puml/puml_graph/graph.py:445: error: Signature of "add_node" incompatible with supertype "Graph"  [override]
tel2puml/puml_graph/graph.py:445: note:      Superclass:
tel2puml/puml_graph/graph.py:445: note:          def add_node(self, node_for_adding: Any, **attr: Any) -> None
tel2puml/puml_graph/graph.py:445: note:      Subclass:
tel2puml/puml_graph/graph.py:445: note:          def add_node(self, node: PUMLNode) -> None
tel2puml/puml_graph/graph.py:456: error: Signature of "add_edge" incompatible with supertype "Graph"  [override]
tel2puml/puml_graph/graph.py:456: note:      Superclass:
tel2puml/puml_graph/graph.py:456: note:          def add_edge(self, u_of_edge: Any, v_of_edge: Any, **attr: Any) -> None
tel2puml/puml_graph/graph.py:456: note:      Subclass:
tel2puml/puml_graph/graph.py:456: note:          def add_edge(self, start_node: PUMLNode, end_node: PUMLNode) -> None
tel2puml/puml_graph/graph.py:465: error: Incompatible types (expression has type "dict[str, Any]", TypedDict item "start_node_attr" has type "NXNodeAttributes")  [typeddict-item]
tel2puml/puml_graph/graph.py:466: error: Incompatible types (expression has type "dict[str, Any]", TypedDict item "end_node_attr" has type "NXNodeAttributes")  [typeddict-item]
tel2puml/puml_graph/graph.py:540: error: Invalid index type "PUMLNode" for "dict[PUMLEventNode | PUMLOperatorNode, list[PUMLEventNode | PUMLOperatorNode]]"; expected type "PUMLEventNode | PUMLOperatorNode"  [index]
tel2puml/puml_graph/graph.py:546: error: Call to untyped function (unknown) in typed context  [no-untyped-call]
tel2puml/puml_graph/graph.py:558: error: Incompatible return value type (got "list[PUMLNode]", expected "list[PUMLEventNode | PUMLOperatorNode]")  [return-value]
tel2puml/puml_graph/graph.py:650: error: "DiGraph[Any]" has no attribute "node_counts"  [attr-defined]
tel2puml/puml_graph/graph.py:651: error: "DiGraph[Any]" has no attribute "branch_counts"  [attr-defined]
tel2puml/puml_graph/graph.py:652: error: "DiGraph[Any]" has no attribute "kill_counts"  [attr-defined]
tel2puml/puml_graph/graph.py:653: error: "DiGraph[Any]" has no attribute "parent_graph_nodes_to_node_ref"  [attr-defined]
tel2puml/puml_graph/graph.py:656: error: Incompatible return value type (got "DiGraph[Any]", expected "PUMLGraph")  [return-value]
tel2puml/loop_detection/calculate_loop_components.py:242: error: Incompatible types in assignment (expression has type "Graph[T]", variable has type "DiGraph[T]")  [assignment]
tests/tel2puml/puml_graph/test_graph.py:49: error: Non-overlapping equality check (left operand type: "dict[str, str]", right operand type: "dict[str, bool]")  [comparison-overlap]
tests/tel2puml/puml_graph/test_graph.py:313: error: Non-overlapping equality check (left operand type: "dict[str, str]", right operand type: "dict[str, bool]")  [comparison-overlap]
tests/tel2puml/puml_graph/test_graph.py:322: error: Non-overlapping equality check (left operand type: "dict[str, str]", right operand type: "dict[str, bool]")  [comparison-overlap]
tests/tel2puml/puml_graph/test_graph.py:535: error: Need type annotation for "node_set" (hint: "node_set: Set[<type>] = ...")  [var-annotated]
tests/tel2puml/puml_graph/test_graph.py:581: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/tel2puml/puml_graph/test_graph.py:582: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/tel2puml/puml_graph/test_graph.py:583: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tests/tel2puml/puml_graph/test_graph.py:608: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/tel2puml/puml_graph/test_graph.py:610: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/tel2puml/puml_graph/test_graph.py:611: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/tel2puml/puml_graph/test_graph.py:613: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tests/tel2puml/puml_graph/test_graph.py:615: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tests/tel2puml/puml_graph/test_graph.py:641: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/tel2puml/puml_graph/test_graph.py:643: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/tel2puml/puml_graph/test_graph.py:644: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/tel2puml/puml_graph/test_graph.py:646: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tests/tel2puml/puml_graph/test_graph.py:648: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:50: error: Incompatible types in assignment (expression has type "None", variable has type "int")  [assignment]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:115: error: No overload variant of "__setitem__" of "list" matches argument types "int", "PUMLNode"  [call-overload]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:115: note: Possible overload variants:
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:115: note:     def __setitem__(self, SupportsIndex, PUMLOperatorNode, /) -> None
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:115: note:     def __setitem__(self, slice, Iterable[PUMLOperatorNode], /) -> None
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:149: error: Incompatible types in assignment (expression has type "list[PUMLOperatorNode | PUMLNode]", variable has type "list[PUMLOperatorNode]")  [assignment]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:154: error: Incompatible return value type (got "tuple[PUMLNode, Node | None]", expected "tuple[PUMLNode, Node]")  [return-value]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:219: error: Argument 2 to "has_event_set_as_subset" has incompatible type "list[str | None]"; expected "list[str]"  [arg-type]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:335: error: Argument 1 to "create_event_node" of "PUMLGraph" has incompatible type "str | None"; expected "str"  [arg-type]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:367: error: Incompatible types in assignment (expression has type "PUMLKillNode", variable has type "PUMLEventNode")  [assignment]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:370: error: Incompatible types in assignment (expression has type "PUMLOperatorNode | PUMLEventNode", variable has type "PUMLEventNode")  [assignment]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:385: error: Incompatible types in assignment (expression has type "PUMLNode", variable has type "PUMLEventNode")  [assignment]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:399: error: Argument 2 to "update_puml_graph_with_event_node" has incompatible type "Node | None"; expected "Node"  [arg-type]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:410: error: Incompatible types in assignment (expression has type "PUMLNode", variable has type "PUMLEventNode")  [assignment]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:421: error: Incompatible types in assignment (expression has type "PUMLOperatorNode", variable has type "PUMLEventNode")  [assignment]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:456: error: Argument 1 to "create_operator_node_pair" of "PUMLGraph" has incompatible type "PUMLOperator | None"; expected "PUMLOperator"  [arg-type]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:466: error: Incompatible return value type (got "tuple[PUMLOperatorNode | PUMLEventNode, Node]", expected "tuple[PUMLOperatorNode, Node]")  [return-value]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:519: error: No overload variant of "__setitem__" of "list" matches argument types "int", "PUMLEventNode"  [call-overload]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:519: note: Possible overload variants:
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:519: note:     def __setitem__(self, SupportsIndex, PUMLOperatorNode, /) -> None
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:519: note:     def __setitem__(self, slice, Iterable[PUMLOperatorNode], /) -> None
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:521: error: Incompatible return value type (got "tuple[PUMLNode, Node | None]", expected "tuple[PUMLNode, Node]")  [return-value]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:537: error: Argument 2 to "create_logic_merge" of "LogicBlockHolder" has incompatible type "Node | None"; expected "Node"  [arg-type]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:549: error: Argument 3 to "handle_rotate_path" has incompatible type "PUMLNode"; expected "PUMLOperatorNode"  [arg-type]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:576: error: Incompatible types in assignment (expression has type "PUMLNode", variable has type "PUMLOperatorNode")  [assignment]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:583: error: Incompatible types in assignment (expression has type "PUMLOperatorNode | PUMLEventNode", variable has type "PUMLOperatorNode")  [assignment]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:643: error: Incompatible return value type (got "tuple[PUMLNode, Node]", expected "tuple[PUMLOperatorNode | PUMLEventNode, Node]")  [return-value]
tel2puml/walk_puml_graph/walk_puml_logic_graph.py:683: error: Incompatible types in assignment (expression has type "PUMLEventNode", variable has type "PUMLOperatorNode")  [assignment]
tel2puml/puml_graph/graph_loop_extract.py:56: error: Need type annotation for "start_loop_nodes" (hint: "start_loop_nodes: Set[<type>] = ...")  [var-annotated]
tel2puml/puml_graph/graph_loop_extract.py:57: error: Need type annotation for "end_loop_nodes" (hint: "end_loop_nodes: Set[<type>] = ...")  [var-annotated]
tests/tel2puml/walk_puml_graph/test_walk_puml_logic_graph.py:15: error: Module "tel2puml.puml_graph.graph" does not explicitly export attribute "PUMLOperatorNodes"  [attr-defined]
tests/tel2puml/walk_puml_graph/test_walk_puml_logic_graph.py:225: error: Argument 1 to "rotate_path" of "LogicBlockHolder" has incompatible type "Node | None"; expected "Node"  [arg-type]
tests/tel2puml/puml_graph/test_graph_loop_extract.py:58: error: Argument 1 to "remove" of "list" has incompatible type "Hashable"; expected "str"  [arg-type]
tests/tel2puml/puml_graph/test_graph_loop_extract.py:64: error: Argument 1 to "remove" of "list" has incompatible type "Hashable"; expected "str"  [arg-type]
tests/tel2puml/puml_graph/test_graph_loop_extract.py:75: error: Argument 1 to "remove" of "list" has incompatible type "Hashable"; expected "str"  [arg-type]
tests/tel2puml/puml_graph/test_graph_loop_extract.py:81: error: Argument 1 to "remove" of "list" has incompatible type "Hashable"; expected "str"  [arg-type]
tests/tel2puml/puml_graph/test_graph_loop_extract.py:303: error: Argument 1 to "remove" of "list" has incompatible type "tuple[Hashable, Hashable]"; expected "tuple[tuple[str, str, int], tuple[str, str, int]]"  [arg-type]
tests/tel2puml/puml_graph/test_graph_loop_extract.py:321: error: Argument 1 to "check_correct_start_and_end_nodes" of "TestGetLoopStartAndEnd" has incompatible type "list[tuple[PUMLNode, PUMLNode]]"; expected "list[tuple[PUMLEventNode, PUMLEventNode]]"  [arg-type]
tests/tel2puml/puml_graph/test_graph_loop_extract.py:336: error: Argument 1 to "check_correct_start_and_end_nodes" of "TestGetLoopStartAndEnd" has incompatible type "list[tuple[PUMLNode, PUMLNode]]"; expected "list[tuple[PUMLEventNode, PUMLEventNode]]"  [arg-type]
tel2puml/puml_graph/graph_loop_insert.py:42: error: Argument 1 to "insert_loops" has incompatible type "PUMLGraph | None"; expected "PUMLGraph"  [arg-type]
tests/tel2puml/puml_graph/test_graph_loop_insert.py:47: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/tel2puml/puml_graph/test_graph_loop_insert.py:57: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/tel2puml/puml_graph/test_graph_loop_insert.py:73: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/tel2puml/puml_graph/test_graph_loop_insert.py:79: error: Item "None" of "PUMLGraph | None" has no attribute "subgraph_nodes"  [union-attr]
tests/tel2puml/puml_graph/test_graph_loop_insert.py:95: error: Argument 1 to "check_networkx_graph_equivalence" has incompatible type "PUMLGraph | None"; expected "DiGraph[Any]"  [arg-type]
tests/tel2puml/puml_graph/test_graph_loop_insert.py:100: error: Argument 1 to "check_networkx_graph_equivalence" has incompatible type "PUMLGraph | None"; expected "DiGraph[Any]"  [arg-type]
tel2puml/pv_to_puml.py:187: error: Returning Any from function declared to return "list[PVEvent]"  [no-any-return]
tel2puml/pv_to_puml.py:282: error: Argument 1 to "cluster_events_by_job_id" has incompatible type "Generator[list[PVEvent], Any, None]"; expected "Iterable[PVEvent]"  [arg-type]
Found 174 errors in 22 files (checked 62 source files)
